python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -0,0 +1,923 @@
+[[X]] `type(True)`
+***
+<div class = "answer">
+These are all examples of valid Python code! `print(type(4))` returns the type of the argument `4` (which is `int`, short for integer) and prints that output to the screen. `"python.upper()"` makes the string "python" all uppercase. `type("4.3")` might be a little tricky, since 4.3 is a decimal number (or **float**) but it's in quotation marks-- but those quotation marks just mean that, in this case, "4.3" is a string! `type(True)` might also be tricky, since at first it looks like a string that doesn't have the required quotation marks. However, `True` and `False` are special in Python, and have the type `boolean`.  
+</div>
+***
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+print(x)
+
+</script>
+</lia-keep>
+</div>
+
+## Variables
+
+Rather than passing a value (like a string or a number) directly to a function, we can also **assign** values to variables and pass those to functions. **Variables** are ways to store values and objects for use later.
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+my_num = 17
+print(my_num)
+
+</script>
+</lia-keep>
+</div>
+
+While the above example may seem trivial, variables are extremely useful for re-using values and objects, for making your code easier to edit (if you store a value in a variable and then need to change it later, you only need to change the value in the variable definition, rather than every time in your code you've used that value), and making your code more readable.
+
+<div class = "important">
+Tip: It's a good idea to use descriptive names for your variables! Python won't care if you decide to assign a value to the variable `x`, but it's helpful to anyone reading your code (including yourself in 6 months!) to have your variable names describe the value that it holds.
+</div>
+
+@sage
+<div class="python">
+<lia-keep>
+<script type="text/x-sage">
+
+dogs = 3
+cats = 5
+all_pets = dogs + cats
+print(all_pets)
+</script>
+</lia-keep>
+</div>
+
+In the above example, we have defined variables to contain the number of dogs and the number of cats, and then used these variables in a calculation to get the total number of pets, which is itself assigned to a variable. The `all_pets` variable can then be used repeatedly later (maybe we want to calculate costs associated with feeding and housing all of these pets, for example, or how many collars we'll need). Now, if the number of dogs or cats changes, we can just change the value in the variable statement and re-run our code!
+
+<div class = "important">
+While strings in Python need to be in quotation marks, variable names do not. The reason that strings require quotation marks is so that Python knows that they aren't variables! Numbers and boolean values like `True` and `False` do not need quotation marks either, but you also cannot use numbers or boolean values by themselves as variable names (a mix of letters and number is okay).
```

Comment:
```suggestion
While strings in Python need to be in quotation marks, variable names do not. The reason that strings require quotation marks is so that Python knows that when you type "abc", you mean the **literal characters** "abc", and not some variable with the **name** `abc`.  

Numbers and boolean values like `True` and `False` do not need quotation marks either, but you also cannot use numbers or boolean values by themselves as variable names (but naming a variable using a mix of letters and number is okay).
```

Reply:
Much better phrasing, ty!